src/include/stir/recon_buildblock/LogcoshPrior.h

```diff
@@ -224,18 +225,19 @@ class LogcoshPrior:  public
       { return tanh(x)/x; }
     }
 
-    //! The Hessian of log(cosh()) is sech^2(x) = (1/cosh(x))^2
+    //! The second partial derivatives of the LogCosh Prior
     /*!
-     This function returns the hessian of the logcosh function
-     * @param d the difference between the ith and jth voxel.
-     * @param scalar is the logcosh scalar value controlling the priors transition between the quadratic and linear behaviour
-     * @return the second derivative of the log-cosh function
+     Diagonal refers to the second derivative w.r.t. x_j only (i.e. diagonal of the Hessian)
+     Off-diagonal refers to the second derivative w.r.t. x_j and x_k (i.e. off-diagonal of the Hessian)
+     For LogCosh, the off diagonal is the negative of the diagonal.
+     * @param x_j is the target voxel.
+     * @param x_k is the voxel in the neighbourhood.
+     * @return the second order partial derivatives of the LogCosh Prior
      */
-    static inline float Hessian(const float d, const float scalar)
-    {
-      const float x = d * scalar;
-      return square((1/ cosh(x)));
-    }
+    //@{
+    float diagonal_second_derivative(const float x_j, const float x_k) const;
+    float off_diagonal_second_derivative(const float x_j, const float x_k) const;
```

Comment:
I believe `float` has to be `elemT`

Reply:
I believe we can do this in all instances and for all priors? 
```
    elemT diagonal_second_derivative(const elemT x_j, const elemT x_k) const;
    elemT off_diagonal_second_derivative(const elemT x_j, const elemT x_k) const;
```